packages/sanity/src/desk/components/pane/PaneLayout.tsx

```diff
@@ -22,6 +22,7 @@ export function PaneLayout(
     CardProps &
     Omit<React.HTMLProps<HTMLDivElement>, 'as' | 'height' | 'ref' | 'wrap'>
 ) {
+  const {forceUpdate: forcePopoverUpdate} = usePopover()
```

Comment:
Are we sure this is not causing any perf. issues and it only re-renders when strictly necessary?

Reply:
As far as I can tell, it wouldn't. There is two parts to this: the context value, and the hook that listens to updates.
Whenever you call `forceUpdate`, it will set the context value with a new random string. The [`useOnForcedUpdate `](https://github.com/sanity-io/ui/pull/1099/files#diff-c0c8b59e352a97f55536df16b5ef9f4d5f333c9171e1eea39ab2dc26ae4c5a72) hook will listen to that change, and make sure to use the `update()` method to recalculate the position. It will trigger a re-render of the popover component itself, and there might be "unnecessary" re-renders if the panes state change a lot, but this can't be avoided. It shouldn't trigger any re-renders if any popovers isn't open.

I have a branch locally where I have a slightly debounced version that runs every 15ms, that also fixes the other issue with the annotation tooltip. I'll push this, and we can test and see if we are happy.